scanspec/specs.py

```diff
@@ -385,23 +389,25 @@ def _line_from_indexes(self, indexes: np.ndarray) -> Dict[Any, np.ndarray]:
         # self.start is the first centre point, but we need the lower bound
         # of the first point as this is where the index array starts
         first = self.start - step / 2
-        return {self.key: indexes * step + first}
+        return {self.axis: indexes * step + first}
 
     def create_dimensions(self, bounds=True, nested=False) -> List[Dimension]:
         return _dimensions_from_indexes(
-            self._line_from_indexes, self.keys(), self.num, bounds
+            self._line_from_indexes, self.axes(), self.num, bounds
         )
 
     @alternative_constructor
     def bounded(
-        key: Annotated[Any, schema(description="An identifier for what to move")],
+        axis: Annotated[Any, api_schema(description="An identifier for what to move")],
```

Comment:
I think we should turn all `axis` types to `str`

Reply:
Agreed, all instances of axis have been updated and have had their unit tests adapted to match